vm/src/bench/kernel.rs

```diff
@@ -0,0 +1,421 @@
+        flags: SendFlags,
+    ) -> fvm::kernel::Result<SendResult> {
+        self.inner_kernel.send::<BenchKernel<C>>(recipient, method, params, value, gas_limit, flags)
+    }
+}
+
+impl<C> ActorOps for BenchKernel<C>
+where
+    C: CallManager,
+{
+    fn resolve_address(&self, address: &Address) -> Result<ActorID> {
+        self.inner_kernel.resolve_address(address)
+    }
+
+    fn get_actor_code_cid(&self, id: ActorID) -> Result<Cid> {
+        self.inner_kernel.get_actor_code_cid(id)
+    }
+
+    fn next_actor_address(&self) -> Result<Address> {
+        self.inner_kernel.next_actor_address()
+    }
+
+    fn create_actor(
+        &mut self,
+        code_id: Cid,
+        actor_id: ActorID,
+        delegated_address: Option<Address>,
+    ) -> Result<()> {
+        self.inner_kernel.create_actor(code_id, actor_id, delegated_address)
+    }
+
+    fn get_builtin_actor_type(&self, code_cid: &Cid) -> Result<u32> {
+        self.inner_kernel.get_builtin_actor_type(code_cid)
+    }
+
+    fn get_code_cid_for_type(&self, typ: u32) -> Result<Cid> {
+        self.inner_kernel.get_code_cid_for_type(typ)
+    }
+
+    fn balance_of(&self, actor_id: ActorID) -> Result<TokenAmount> {
+        self.inner_kernel.balance_of(actor_id)
+    }
+
+    fn lookup_delegated_address(&self, actor_id: ActorID) -> Result<Option<Address>> {
+        self.inner_kernel.lookup_delegated_address(actor_id)
+    }
+}
+
+impl<C> IpldBlockOps for BenchKernel<C>
+where
+    C: CallManager,
+{
+    fn block_open(&mut self, cid: &Cid) -> std::result::Result<(u32, BlockStat), ExecutionError> {
+        self.inner_kernel.block_open(cid)
+    }
+
+    fn block_create(&mut self, codec: u64, data: &[u8]) -> Result<BlockId> {
+        self.inner_kernel.block_create(codec, data)
+    }
+
+    fn block_link(&mut self, id: BlockId, hash_fun: u64, hash_len: u32) -> Result<Cid> {
+        self.inner_kernel.block_link(id, hash_fun, hash_len)
+    }
+
+    fn block_read(&self, id: BlockId, offset: u32, buf: &mut [u8]) -> Result<i32> {
+        self.inner_kernel.block_read(id, offset, buf)
+    }
+
+    fn block_stat(&self, id: BlockId) -> Result<BlockStat> {
+        self.inner_kernel.block_stat(id)
+    }
+}
+
+impl<C> CircSupplyOps for BenchKernel<C>
+where
+    C: CallManager,
+{
+    // Not forwarded. Circulating supply is taken from the TestData.
+    fn total_fil_circ_supply(&self) -> Result<TokenAmount> {
+        Ok(self.test_config.circ_supply.clone())
+    }
+}
+
+impl<C> CryptoOps for BenchKernel<C>
+where
+    C: CallManager,
+{
+    // forwarded
+    fn hash(&self, code: u64, data: &[u8]) -> Result<MultihashGeneric<64>> {
+        self.inner_kernel.hash(code, data)
+    }
+
+    // forwarded
+    fn compute_unsealed_sector_cid(
+        &self,
+        proof_type: RegisteredSealProof,
+        pieces: &[PieceInfo],
+    ) -> Result<Cid> {
+        self.inner_kernel.compute_unsealed_sector_cid(proof_type, pieces)
+    }
+
+    // forwarded
```

Comment:
We'll probably need to fake this at some point, but if this is working without doing so, all good.